src/NServiceBus.Core/Recoverability/Faults/ErrorQueueSettings.cs

```diff
@@ -23,19 +29,31 @@ public static string ErrorQueueAddress(this ReadOnlySettings settings)
         {
             string errorQueue;
 
-            if (settings.TryGet("errorQueue", out errorQueue))
+            if (TryGetErrorQueueAddress(settings, out errorQueue))
             {
-                Logger.Debug("Error queue retrieved from code configuration via 'EndpointConfiguration.SendFailedMessagesTo().");
                 return errorQueue;
             }
 
+            return DefaultErrorQueueName;
+        }
+
+        internal static bool TryGetErrorQueueAddress(this ReadOnlySettings settings, out string errorQueue)
+        {
+            if (settings.HasExplicitValue(SettingsKey))
+            {
+                Logger.Debug("Error queue retrieved from code configuration via 'EndpointConfiguration.SendFailedMessagesTo().");
+                errorQueue = settings.Get<string>(SettingsKey);
+                return true;
+            }
+
             var section = settings.GetConfigSection<MessageForwardingInCaseOfFaultConfig>();
             if (section != null)
             {
                 if (!string.IsNullOrWhiteSpace(section.ErrorQueue))
                 {
                     Logger.Debug("Error queue retrieved from <MessageForwardingInCaseOfFaultConfig> element in config file.");
-                    return section.ErrorQueue;
+                    errorQueue = section.ErrorQueue;
+                    return true;
```

Comment:
Not sure I follow, you mean to only tell users to use the code first version to prep them for v7 where this will be the only option?